FiveCalls/FiveCalls/Operator.swift

```diff
@@ -23,39 +23,43 @@ class Operator {
         queue = .main
     }
 
-    func fetchIssues(delegate: AppStateDelegate, completion: @escaping (OperationResult) -> Void) {
+    func fetchIssues(setIssues: @MainActor @escaping ([Issue]) -> ()) {
         let operation = FetchIssuesOperation()
         operation.completionBlock = { [weak operation] in
             if let issues = operation?.issuesList {
                 DispatchQueue.main.async {
-                    delegate.setIssues(issues: issues)
-                    completion(.success)
+                    setIssues(issues)
                 }
             } else if let error = operation?.error {
                 print("Could not load issues: \(error.localizedDescription)..")
                 
-                DispatchQueue.main.async {
-                    print(error.localizedDescription)
-                    
-                    if error.isOfflineError() {
-                        completion(.offline)
-                    } else {
-                        completion(.serverError(error))
-                    }
-                }
+                // TODO(nickoneill): do we need to return an error? does it make more sense to have a function that sets an error in app state? or make this async with a completion so state in the component can handle the error?
```

Comment:
Completions seem to be getting phased out in lieu of async/await and setting error in the app state would be the SwiftUI-y way to do it.

Reply:
resolved with the unidirectional data flow